### ISSUE TITLE
Support round-tripping 64-bit ints

### ISSUE THREAD

**Issue by codeinred** [stephenberry/glaze#64]
**Description**

Support round-tripping 64-bit ints. Consider the following struct:

```cpp
struct Value {
    long long value;
};
template <>
struct glz::meta<Value> {
    using T = Value;
    constexpr static auto value = object(
        "value", &Value::value
    );
};
```

My general expectation is that, provided v doesn't contain any floating point numbers, the following holds:

```cpp
glz::read_json<Value>(glz::write_json(v)) == v
```

However, it seems as though numbers are always parsed as floats, even if the corresponding field expects an int.

Consider the following example:

```cpp
int main() {
    Value v1 { 32948729483739289 };

    std::string json = glz::write_json(v1);

    Value v2 = glz::read_json<Value>(json);

    printf("Json: %s\n", glz::prettify(json).c_str());

    if (v1.value != v2.value) {
        printf("Values not equal. v1 = %lli, v2 = %lli\n", v1.value, v2.value);
        return 1;
    } else {
        puts("All good!");
        return 0;
    }
}
```

This value is serialized properly (all digits intact), but when it's parsed, it's parsed as a float, then converted to an int.

This is the output of the above:
```
Json: {
   "value": 32948729483739289
}
Values not equal. v1 = 32948729483739289, v2 = 32948729483739288
```

Would it be possible to parse numbers as integers if the corresponding field calls for it?

**Comment by codeinred** [comment]
Addendum: I just want to mention that this is particularly relevant in cases where nanosecond timestamps or integral numeric IDs are being stored in json. Nanosecond timestamps allow a bit of fuzzyness (if the timestamp is off by a microsecond, that's often not the end of the world), but 64-bit integral IDs need to match exactly

**Comment by mwalcott3** [comment]
Yeah number parsing is going to change soonish (sometime next week). This will get fixed. Right now I would expect only values between -2^53 to 2^53 would work properly. Values were originally parsed in as floating point numbers because we wanted to allow stuff like 1.7e10 which is handled by default in float parsers but typically not int parsers.

**Comment by mwalcott3** [comment]
Fixed in 6e7cab33b7ded72431c55a51481e8003f32efa5d
